framework/src/loops/ComputeElemAuxBcsThread.C

```diff
@@ -79,8 +80,20 @@ ComputeElemAuxBcsThread::operator()(const ConstBndElemRange & range)
             needed_mat_props.insert(mp_deps.begin(), mp_deps.end());
           }
           _problem.setActiveMaterialProperties(needed_mat_props, _tid);
+
+          const Elem * neighbor = elem->neighbor_ptr(side);
+
           _problem.reinitMaterialsFace(elem->subdomain_id(), _tid);
+          if (neighbor != nullptr)
```

Comment:
Ideally we need to reinitialize material on the neighbour side to correctly get update value for  an interface material using getNeighborMatarialProperty(). This is the reason for this change, however it doesn't work

Reply:
@lindsya, @jiangwen84  and @andrsd  I'm having some big trouble to make this portion of the code to work correctly (this was not filling before some). This works fine for standard build but crashes on debug build. I believe some the of the neighbor pointer return some nonsense for the interface material in debug mode. I can fix it in sept mode (the posted code), but I can't fix it for `dbg` mode. any idea on how to fix this?

Reply:
Well if you were referring to this:
```
          if (neighbor != nullptr)
            if (neighbor->active())
            {
              SwapBackSentinel neighbor_sentinel(
                  _problem, &FEProblem::swapBackMaterialsNeighbor, _tid);
              _problem.reinitNeighbor(elem, side, _tid);
            }
```
`FEProblemBase::reinitNeighbor` does not touch materials. Yo probably wanted `FEProblemBase::reinitMaterialsNeighbor`

Reply:
@lindsayad Rebasing this branch now gives a bunch of conflicts when rebasing.
If you are actively  working on this I I'll leave it to you :)